pkg/rollup/rollup.go

```diff
@@ -0,0 +1,133 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package rollup
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/timescale/promscale/pkg/util"
+)
+
+type DownsampleResolution struct {
```

Comment:
Ok I see you have defined the struct. I'd recommend using it in config struct instead of custom string format.